apps/emqx_authn/src/simple_authn/emqx_authn_redis.erl

```diff
@@ -141,16 +141,22 @@ authenticate(
                 {ok, []} ->
                     ignore;
                 {ok, Values} ->
-                    Selected = merge(Fields, Values),
-                    case
-                        emqx_authn_utils:check_password_from_selected_map(
-                            Algorithm, Selected, Password
-                        )
-                    of
-                        ok ->
-                            {ok, emqx_authn_utils:is_superuser(Selected)};
-                        {error, _Reason} ->
-                            ignore
+                    case check_query_matched(Values) of
+                        false ->
+                            %% key not exists
+                            ignore;
+                        _ ->
+                            Selected = merge(Fields, Values),
```

Comment:
When `Values` is `undefined` or a list of `undefined`s, this merge result is an empty map.
To avoid duplicating the `Values` handing, we can just check if `Selected =:= #{}`